packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -99,76 +89,47 @@ export class EVMAAWallet extends LoggerWrapper {
-                    }
-                    case "sft": {
-                        tokenId = evmToken.tokenId;
-                        const { request } = await publicClient.simulateContract({
-                            account: this.account,
-                            address: contractAddress,
-                            abi: erc1155,
-                            functionName: "safeTransferFrom",
-                            args: [this.getAddress(), toAddress, tokenId, (config as SFTTransferType).quantity, "0x00"],
-                            ...(usesGelatoBundler(this.chain) && gelatoBundlerProperties),
-                        });
-                        transaction = await publicClient.writeContract(request);
-                        break;
-                    }
-                    case "nft": {
-                        tokenId = evmToken.tokenId;
-                        const { request } = await publicClient.simulateContract({
-                            account: this.account,
-                            address: contractAddress,
-                            abi: erc721,
-                            functionName: "safeTransferFrom",
-                            args: [this.getAddress(), toAddress, tokenId],
-                            ...(usesGelatoBundler(this.chain) && gelatoBundlerProperties),
-                        });
-                        transaction = await publicClient.writeContract(request);
-                        break;
-                    }
-                    default: {
-                        throw new WalletSdkError(`Token not supported`);
-                    }
-                }
+            if (!isAddress(toAddress)) {
+                throw new Error(`Invalid recipient address: '${toAddress}' is not a valid EVM address.`);
+            }
 
-                if (transaction != null) {
-                    return transaction;
-                } else {
-                    throw new TransferError(
-                        `Error transferring token ${evmToken.contractAddress}
-                    ${!tokenId ? "" : ` tokenId=${tokenId}`}
-                    ${!transaction ? "" : ` with transaction hash ${transaction}`}`
-                    );
-                }
+            if (!isAddress(config.token.contractAddress)) {
+                throw new Error(
+                    `Invalid contract address: '${config.token.contractAddress}' is not a valid EVM address.`
+                );
+            }
+
+            const tx = {
+                ...transferParams({
+                    contract: config.token.contractAddress,
+                    to: toAddress,
+                    from: this.account,
+                    config,
+                }),
+            };
+
+            try {
+                const client = this.smartAccountClient.extend(publicActions);
+                const { request } = await client.simulateContract(tx);
+                return client.writeContract(request);
```

Comment:
seeing the omission of 
```
...(usesGelatoBundler(this.chain) && gelatoBundlerProperties),
```
so just making sure we test on astar

Reply:
It's on the shortlist for passkey release QA
Since it's using our wrapped `SmartAccountClient`, where we add these properties on sendTransaction, we should be automatically adding those properties. From the `viem` docs:

> Internally, uses a [Wallet Client](vscode-file://vscode-app/Applications/Cursor.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) to call the [sendTransaction action](vscode-file://vscode-app/Applications/Cursor.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) with [ABI-encoded data](vscode-file://vscode-app/Applications/Cursor.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html).